src/js/components/IndexPage.jsx

```diff
@@ -11,16 +11,21 @@ export class IndexPage extends PureComponent {
 
   // @todo remove forced condition
   // respond to search input
-  handleSubmit() {
-    return this.setState({ showResults: true });
+  handleSearchDone(data, searchTerm) {
+    if (!data.length) return this.setState({ showResults: false });
+    return this.setState({ showResults: true, resultData: data, searchTerm });
   }
 
   render(props) {
     return (
       <div>
-        <SearchBar handleSubmit={() => this.handleSubmit()} />
+        <SearchBar searchDone={(data, searchTerm) => this.handleSearchDone(data, searchTerm)} />
```

Comment:
Yep, I agree on both points after seeing your thinking. `Done` oops, yeah it is like boolean, async-ish terminology - I'll remove that.

Typically, as a personal convention only, I don't add `on` because it visually clashes (maybe confusing later devs possibly coming from Angular, et al) with core React's DOM event handlers implementation. However, I'm cool with merging the code styles of mobile!